component/attr_cache/attr_cache.go

```diff
@@ -868,17 +871,22 @@ func (ac *AttrCache) GetAttr(options internal.GetAttrOptions) (*internal.ObjAttr
 		// Retrieved attributes so cache them
 		// TODO: bug: when cacheDirs is true, the cache limit will cause some directories to be double-listed
 		// TODO: shouldn't this be an LRU? This sure looks like the opposite...
-		if len(ac.cacheMap) < ac.maxFiles {
-			ac.cacheMap[truncatedPath] = newAttrCacheItem(pathAttr, true, time.Now())
+		if pathAttr == nil {
+			ac.cacheMap.insert(&internal.ObjAttr{Path: options.Name}, true, time.Now())
 		} else {
-			log.Debug("AttrCache::GetAttr : %s skipping adding to attribute cache because it is full", options.Name)
+			ac.cacheMap.insert(pathAttr, true, time.Now())
 		}
+
 		if ac.cacheDirs {
 			ac.markAncestorsInCloud(getParentDir(options.Name), time.Now())
 		}
 	} else if err == syscall.ENOENT {
 		// Path does not exist so cache a no-entry item
```

Comment:
Maybe this comment could be made clearer. The goal of this part of the code is to cache the fact that this item does not exist. So we make a new cache entry like normal, but with exists set to false.